# Introduction

The increasing prevalence of social media use among students has raised concerns about its potential effects on academic performance and mental health. Social media platforms are widely used for communication, entertainment, and information sharing, but excessive use may lead to distractions, interpersonal conflict, and symptoms associated with social media addiction. As a result, understanding how patterns of social media use relate to student outcomes has become an important area of study.

This project analyzes a student social media dataset to examine how different patterns of social media use are associated with academic performance and mental health outcomes. The dataset includes variables related to social media behavior, such as average daily usage, most-used platforms, conflicts caused by social media, and whether students perceive themselves as addicted to social media. In addition, demographic characteristics, including age, gender, education level, and country of residence, are included to explore whether these factors influence the relationship between social media use and student outcomes.

Through statistical analysis, this study aims to identify potentially harmful patterns of social media use that may be associated with poorer academic performance and negative mental health indicators. The analysis will also examine whether these relationships differ across demographic groups, such as age, gender, education level, and geographic region.

The primary research question is whether higher levels of social media usage, conflict related to social media, or self-perceived addiction are associated with worse academic performance and poorer mental health among students. It is hypothesized that greater daily social media use and higher indicators of conflict or addiction will be associated with lower academic performance and poorer mental health outcomes. Additionally, it is expected that demographic factors—particularly education level—may moderate these relationships, potentially reducing the strength of harmful associations among students at higher education levels.

**Research Question 1:** Are higher levels of social media use, conflict, or self-perceived addiction associated with worse academic performance and poorer mental health status among students?

**Research Question 2:** Do associations of social media, conflict, and academic performance differ across demographic subgroups (age, gender, education level, and country group)?


In [37]:
#Load Source Code
import data_cleaning as dc
import exploratory_analysis as ea
import modeling

# Dataset Description

Describe Data

In [38]:
#Load Data
import pandas as pd
import numpy as np
df = pd.read_csv('Students Social Media Addiction.csv')
df.describe()

,Student_ID,Age,Avg_Daily_Usage_Hours,Sleep_Hours_Per_Night,Mental_Health_Score,Conflicts_Over_Social_Media,Addicted_Score
count,705.000000,705.000000,705.000000,705.000000,705.000000,705.000000,705.000000
mean,353.000000,20.659574,4.918723,6.868936,6.226950,2.849645,6.436879
std,203.660256,1.399217,1.257395,1.126848,1.105055,0.957968,1.587165
min,1.000000,18.000000,1.500000,3.800000,4.000000,0.000000,2.000000
25%,177.000000,19.000000,4.100000,6.000000,5.000000,2.000000,5.000000
50%,353.000000,21.000000,4.800000,6.900000,6.000000,3.000000,7.000000
75%,529.000000,22.000000,5.800000,7.700000,7.000000,4.000000,8.000000
max,705.000000,24.000000,8.500000,9.600000,9.000000,5.000000,9.000000


# Preprocessing

In [39]:
#NA Removal
df.isna().sum()

Student_ID                      0
Age                             0
Gender                          0
Academic_Level                  0
Country                         0
Avg_Daily_Usage_Hours           0
Most_Used_Platform              0
Affects_Academic_Performance    0
Sleep_Hours_Per_Night           0
Mental_Health_Score             0
Relationship_Status             0
Conflicts_Over_Social_Media     0
Addicted_Score                  0
dtype: int64

There appears to be no null or missing values in the dataset that must be removed. Additionally, after manually examining all 705 respondants, all answers appear to have be consistent and truthfull. 

In [40]:
# Outlier Removal
df_clean = dc.clean_data(df)
df_clean.describe()

,Student_ID,Age,Avg_Daily_Usage_Hours,Sleep_Hours_Per_Night,Mental_Health_Score,Conflicts_Over_Social_Media,Addicted_Score
count,702.000000,702.000000,702.000000,702.000000,702.000000,702.000000,702.000000
mean,353.096866,20.659544,4.913533,6.875926,6.226496,2.850427,6.435897
std,203.541208,1.397793,1.239162,1.116842,1.100504,0.951982,1.575768
min,1.000000,18.000000,2.000000,4.000000,4.000000,0.000000,3.000000
25%,177.250000,19.250000,4.100000,6.000000,5.000000,2.000000,5.000000
50%,352.500000,21.000000,4.800000,6.900000,6.000000,3.000000,7.000000
75%,529.750000,22.000000,5.800000,7.700000,7.000000,4.000000,8.000000
max,705.000000,24.000000,8.300000,9.600000,8.000000,5.000000,9.000000


3 Outliers have been removed based on a 1.5 * IQR method to ensure data accuracy and a roughly normal distribution. 

# Exploratory Analysis

## Summary Statistics

## Social Media Usage vs Academic Performance

## Correlation Between Usage, Addiction, and Mental Health

## Platform Usage Patterns

## Demographic Analysis (Age, Gender, Country)

# Model Building

## Mental Health Regression Model

## Academic Performance Regression Model

## Interaction Effect Regression Model

# Analysis

Explain Analysis

# Conclusion

Explain Conclusion